pkg/metrics/timer/timer_test.go

```diff
@@ -0,0 +1,130 @@
+package timer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func convertToFormat(buf bytes.Buffer) string {
+	output := buf.String()
+
+	outputText := ""
+
+	for _, item := range strings.Split(output, "\n") {
+		var dat map[string]interface{}
+
+		if err := json.Unmarshal([]byte(item), &dat); err != nil {
+			continue
+		}
+		outputText += dat["msg"].(string) + "\n"
+	}
+
+	return outputText
+}
+
+func TestBasicLogSummaryMultipleStages(t *testing.T) {
+	timer := NewTimer()
+
+	timer.StartTimer(TotalTimeElapsed)
+	timer.StartTimer("testStage1")
+	timer.StartTimer("testStage2")
+	timer.StartTimer("testStage3")
+	timer.StartTimer("testStage4")
+
+	// time.Sleep(5 * time.Second)
+
+	timer.StopTimer("testStage1")
+	timer.StopTimer("testStage2")
+	timer.StopTimer("testStage3")
+	timer.StopTimer("testStage4")
+	timer.StopTimer(TotalTimeElapsed)
+
+	timeElapsed := fmt.Sprintf("Time elapsed for each section\n")
+	time1 := fmt.Sprintf("testStage1: %v\n", timer.stageTimes["testStage1"])
+	time2 := fmt.Sprintf("testStage2: %v\n", timer.stageTimes["testStage2"])
+	time3 := fmt.Sprintf("testStage3: %v\n", timer.stageTimes["testStage3"])
+	time4 := fmt.Sprintf("testStage4: %v\n", timer.stageTimes["testStage4"])
+	timeStageElapsed := fmt.Sprintf("Time elapsed: %v\n", timer.stageTimes[TotalTimeElapsed])
+
+	text := timeElapsed + time1 + time2 + time3 + time4 + timeStageElapsed
+
+	textOutput := bytes.Buffer{}
+
+	logger := logrus.New()
+	logger.Out = &textOutput
+	logger.Level = logrus.DebugLevel
+	logger.Formatter = &logrus.JSONFormatter{}
```

Comment:
why the JSONFormatter?

Reply:
With JSONFormatter, it was easier to pick just the message that the logger was producing rather than the TextFormatter where I had to slice the strings to pull it. Could have created a struct and parsed it into the struct but thought JSON was the same as creating a struct so thought of using that.